### PR TITLE
Send mails on order creation

### PR DESCRIPTION
**Pull request by trowik**
moved acknowledged validation to OrderView

### CODE REVIEW

**Comment by trowik** [comment]
@winged could you review the last commit? Will rebase after #810 is merged

File: timed/subscription/templates/notify_accountants_order.html [hunk 1]
```diff
@@ -0,0 +1,17 @@
+<b>Aufladung von {{hours_added}} Stunde/n</b>
```
Comment (winged) [inline timed/subscription/templates/notify_accountants_order.html:1]: I suggest translating the template to english.

Also, nobody is probably ever going to buy a single hour, so no need to use the singular here

File: timed/subscription/views.py [hunk 2]
```diff
@@ -50,6 +56,64 @@ class OrderViewSet(viewsets.ModelViewSet):
...
+        """Override so we can issue emails on creation."""
+        # check if order is acknowledged and created by admin/accountant
+        if (
+            request.method == "POST"
+            and request.data.get("acknowledged")
+            and not (request.user.is_accountant or request.user.is_superuser)
+        ):
+            raise ValidationError("User can not create confirmed orders!")
+
+        template_txt = get_template("notify_accountants_order.txt")
```
Comment (winged) [inline timed/subscription/views.py:69]: Refactor: I suggest moving the whole emailing code into it's own module or at least a separate function/method.

**Review by winged — CHANGES_REQUESTED** [review]
Some minor changes, otherwise looks good

File: timed/subscription/templates/notify_accountants_order.html (hunk 1 above)
Comment (trowik) [reply timed/subscription/templates/notify_accountants_order.html]: done

File: timed/subscription/views.py (hunk 2 above)
Comment (trowik) [reply timed/subscription/views.py]: done

**Comment by trowik** [comment]
changed the name of env variable for customer-center email

File: timed/subscription/notify_admin.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    project = Project.objects.get(id=request.data.get("project")["id"])
+    customer = project.customer
+
+    duration = datetime.datetime.strptime(request.data.get("duration"), "%H:%M:%S")
+    hours_added = datetime.timedelta(
+        hours=duration.hour, minutes=duration.minute, seconds=duration.second
+    )
+    hours_total = hours_added + project.estimated_time
+
+    subject = f"Customer Center Credits/Reports: {customer.name} hat {hours_added} Stunde/n bestellt."
```
Comment (winged) [inline timed/subscription/notify_admin.py:25]: German text

File: timed/subscription/notify_admin.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
+import datetime
+
+from django.conf import settings
+from django.core.mail import EmailMultiAlternatives, get_connection
+from django.template.loader import get_template, render_to_string
+
+from timed.projects.models import Project
+
+
+def prepare_and_send_email(request):
```
Comment (winged) [inline timed/subscription/notify_admin.py:10]: This is business logic, not request handling. Avoid passing in the request object. Instead, pass in the project model directly. This way, the function can be run from other contexts as well if needed

**Review by winged — CHANGES_REQUESTED** [review]

File: timed/subscription/notify_admin.py (hunk 4 above)
Comment (trowik) [reply timed/subscription/notify_admin.py]: Had to pass order duration as well

**Review by winged — APPROVED** [review]

**Pull request merged**